frontend/test/pytest/test_mid_circuit_measurement.py

```diff
@@ -113,6 +113,32 @@ def circuit(x: float):
 
         assert circuit(jnp.pi)  # m will be equal to True
 
+    def test_with_reset_false(self, backend):
+        """Test measure (reset = False)."""
+
+        @qjit
+        @qml.qnode(qml.device(backend, wires=1))
+        def circuit():
+            qml.Hadamard(wires=0)
+            m1 = measure(wires=0, reset=False, postselect=1)
+            m2 = measure(wires=0)
+            return m1 == m2
+
+        assert circuit()  # both measures are the same
+
+    def test_with_reset_true(self, backend):
+        """Test measure (reset = True)."""
+
+        @qjit
+        @qml.qnode(qml.device(backend, wires=1))
+        def circuit():
+            qml.Hadamard(wires=0)
+            m1 = measure(wires=0, reset=True, postselect=1)
+            m2 = measure(wires=0)
+            return m1 != m2
```

Comment:
This is clever and it does correctly test the functionality, but does rely on the correctness of a second feature to do so. We could consider an alternative that just checks the state final state if we wanted to avoid relying on the correctness of postselect, but I'm also happy to merge as is.

Reply:
Which alternative would be that? I am curious

Reply:
One could check the state for example:

```py
    qml.PauliX(0)  # create |1> state
    measure(wires=0, reset=True)  # reset to |0>
    return qml.state()

assert np.allclose(circuit(), [1., 0.])
```

or the expectation value:

```py
    qml.PauliX(0)  # create |1> state
    measure(wires=0, reset=True)  # reset to |0>
    return qml.expval(qml.PauliZ(0))

assert circuit() == 1.
```